src/transformers/models/bridgetower/modeling_bridgetower.py

```diff
@@ -0,0 +1,1208 @@
+            image_embeds_ = (
+                self.cross_modal_image_transform(self.vit_model.visual.forward_post(image_embeds))
+                + image_token_type_embeddings
+            )
+
+            text_link_tower = self.cross_modal_text_link_tower[link_layer_index]
+            image_link_tower = self.cross_modal_image_link_tower[link_layer_index]
+
+            x1_ = text_link_tower(
+                self.cross_modal_text_transform(text_embeds) + text_token_type_embeddings, x1, extend_text_masks
+            )
+            if irtr_len > 0:
+                y1_ = image_link_tower(
+                    image_embeds_.unsqueeze(1).expand(_bs, irtr_len, _L, _D).reshape(-1, _L, _D),
+                    y1,
+                    extend_image_masks,
+                )
+            else:
+                y1_ = image_link_tower(image_embeds_, y1, extend_image_masks)
+
+            x1 = self.cross_modal_text_layers[link_layer_index + 1](x1_, y1_, extend_text_masks, extend_image_masks)[0]
+            y1 = self.cross_modal_image_layers[link_layer_index + 1](y1_, x1_, extend_image_masks, extend_text_masks)[
+                0
+            ]
+
+            link_layer_index += 1
+
+        text_feats, image_feats = x1, y1
+        cls_feats = self.get_cls_feats(text_feats, image_feats)
+
+        if not return_dict:
+            return tuple(v for v in [text_feats, image_feats, cls_feats] if v is not None)
+
+        return BridgeTowerModelOutput(
+            text_feats=text_feats,
+            image_feats=image_feats,
+            pooler_output=cls_feats,
+        )
+
+    def get_cls_feats(self, text_feats, image_feats):
+        cls_feats_text = self.cross_modal_text_pooler(text_feats)
+        if self.is_clip:
+            cls_feats_image = self.cross_modal_image_pooler(image_feats)
+        else:
+            avg_image_feats = self.avgpool(image_feats.transpose(1, 2)).view(image_feats.size(0), 1, -1)
+            cls_feats_image = self.cross_modal_image_pooler(avg_image_feats)
+        return torch.cat([cls_feats_text, cls_feats_image], dim=-1)
+
+
+class LinkTower(nn.Module):
+    def __init__(self, config, tokenizer_config):
+        super(LinkTower, self).__init__()
+        self.link_tower_type = config.link_tower_type
+        self.hidden_size = config.hidden_size
+        if config.link_tower_type in [
+            "add",
+            "scaled_add",
+            "scaled_add_tensor",
+            "interpolate",
+            "interpolate_tensor",
+            "gated_interpolate",
+            "elementwise_product",
+            "elementwise_product_residual",
+            "linear_add_fpn",
+            "linear_concat",
+            "linear_add",
+            "linear_concat_residual",
+            "linear_add_residual",
+            "mlp_concat",
+            "mlp_add",
+            "mlp_concat_residual",
+            "mlp_add_residual",
+        ]:
```

Comment:
I can see that 2 models have been added to the Hub, therefore only 2 values of this condition are relevant no? (I see that `beta`, `gate` and `dense` are actual modules)
I would advocate to support for now only the `link_tower_type` that are relevant to the ones available on the Hub, and we can have follow-up PRs in the future to support more `link_tower_type`

Reply:
Handled in the updated commits. Cleaned up the code. We plan to release some more models soon too.